packages/core/core/src/services/document-service/transform/id-transform.ts

```diff
@@ -0,0 +1,78 @@
+import { Common } from '@strapi/types';
+import { mapAsync, pipeAsync } from '@strapi/utils';
+import { isObject } from 'lodash/fp';
+import { createIdMap } from './id-map';
+import { extractDataIds as extractDataRelationIds } from './relations/extract/data-ids';
+import { transformDataIdsVisitor as transformRelationDataIds } from './relations/transform/data-ids';
+import { transformOutputIds as transformRelationOutputIds } from './relations/transform/output-ids';
+import { switchIdForDocumentId } from './utils';
+
+/**
+ * Transform input of a query to map document ids to entity ids.
+ */
+async function transformParamsDocumentId(
+  uid: Common.UID.Schema,
+  input: { data?: any; fields?: any; filters?: any; populate?: any; sort?: any },
+  opts: {
+    locale?: string | null;
+    isDraft: boolean;
+  }
+) {
+  const idMap = createIdMap({ strapi });
+
+  // Extract any relation ids from the input
+  if (input.data) {
+    await extractDataRelationIds(idMap, input.data, { ...opts, uid });
+  }
+
+  // Load any relation the extract methods found
+  await idMap.load();
+
+  // Transform any relation ids to entity ids
+  let data = input.data;
+  if (input.data) {
+    data = await transformRelationDataIds(idMap, input.data, { ...opts, uid });
+  }
+
+  // TODO: Transform fields
+  const fields = input.fields;
+  // TODO: Transform filters
+  const filters = input.filters;
```

Comment:
do you plan to do that in this PR or another one ?

Reply:
another one, probably jamie will take care of those!